trustkit/src/main/java/com/datatheorem/android/trustkit/pinning/RootTrustManager.java

```diff
@@ -0,0 +1,51 @@
+package com.datatheorem.android.trustkit.pinning;
+
+import android.net.http.X509TrustManagerExtensions;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.RequiresApi;
+
+import com.datatheorem.android.trustkit.TrustKit;
+import com.datatheorem.android.trustkit.config.DomainPinningPolicy;
+
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+
+import javax.net.ssl.X509TrustManager;
+
+/**
+ * {@link X509TrustManager} used for Certificate Pinning.
+ *
+ * <p>This trust manager delegates to the appropriate {@link PinningTrustManager} decided by the
+ * hostname set by the {@link PinningInterceptor}.</p>
+ */
+@RequiresApi(api = 17)
+public class RootTrustManager implements X509TrustManager {
+    private final ThreadLocal<String> mServerHostname = new ThreadLocal<>();
+
+    @Override
+    public void checkClientTrusted(X509Certificate[] chain, String authType) throws CertificateException {
+        TrustKit.getInstance().getTrustManager(mServerHostname.get()).checkClientTrusted(chain, authType);
+    }
+
+    @Override
+    public void checkServerTrusted(X509Certificate[] chain, String authType) throws CertificateException {
+        String host = mServerHostname.get();
+        DomainPinningPolicy serverConfig =
+                TrustKit.getInstance().getConfiguration().getPolicyForHostname(host);
+        if (serverConfig == null) {
```

Comment:
What's the need for the if/else here? What about `TrustKit.getInstance().getTrustManager(host).checkServerTrusted(chain, authType)` in all cases?

Reply:
Added a comment to address the need